### PR TITLE
Gen 5: Reveal Mail in team preview

### PR DESCRIPTION
**Pull request by Karthik99999**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/page-58#post-8882546

Client Side: smogon/pokemon-showdown-client#1825

### CODE REVIEW

**Comment by DaWoblefet** [comment]
Somewhat related, there was a semi-popular suggestion in the suggestions thread for unique mail: https://www.smogon.com/forums/threads/different-types-of-mail.3685590/

I'm not sure whether or not we want to approve that, but regardless, we might want to future-proof by just checking if the item contains the word 'Mail' or something like that.

File: data/mods/gen7/rulesets.ts [hunk 1]
```diff
@@ -36,11 +36,10 @@ export const Rulesets: {[k: string]: ModdedFormatData} = {
 			this.add('clearpoke');
 			for (const pokemon of this.getAllPokemon()) {
 				const details = pokemon.details.replace(', shiny', '')
-					.replace(/(Arceus|Gourgeist|Genesect|Pumpkaboo|Silvally|Urshifu)(-[a-zA-Z?-]+)?/g, '$1-*');
+					.replace(/(Arceus|Gourgeist|Genesect|Pumpkaboo|Silvally)(-[a-zA-Z?-]+)?/g, '$1-*');
```
Comment (Zarel) [inline data/mods/gen7/rulesets.ts:39]: Honestly I would leave Urshifu in.

```suggestion
					.replace(/(Arceus|Gourgeist|Genesect|Pumpkaboo|Silvally|Urshifu)(-[a-zA-Z?-]+)?/g, '$1-*');
```

We do support using future-gen pokemon in past games as long as it's not too much work.

**Comment by Zarel** [comment]
@Lusamine was kind enough to send us the real Mail icon from the games; let's use it:

mail

**Comment by Zarel** [comment]
> @Lusamine was kind enough to send us the real Mail icon from the games; let's use it:

Oh whoops I thought this was the client PR. Oh well, I'm glad it was used successfully.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**